### PR TITLE
Fix (Core\OpCode) HandleFarSightOpcode

### PR DESCRIPTION
**Pull request by acidmanifesto**
This will process the far sight packet as a is not thread-safe - process it in World::UpdateSessions() compared to the former packet is thread-safe - process it in Map::Update().
This matches up with tc in everything we have with the HandleFarSightOpcode.

## Changes Proposed:
-  Change from Thread-safe to Thread Unsafe so it is handled in the World::Update Sessions instead of the Map:Update
- Changed upa few things and remove a few unneed things as well.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9321

## SOURCE:

TC's HandleFarSightOpcode

## Tests Performed:

- Builds
- Plays

## How to Test the Changes:

1. Create a few dozen new characters and cancel the cinematic.
2. Observice if you do not move with each new character after first login.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nefertumm — APPROVED** [review]

**Comment by IntelligentQuantum** [comment]
@acidmanifesto still bug

**Comment by acidmanifesto** [comment]
> @acidmanifesto still bug

Made some further adjustmetns if you are willing to test one more time.

**Comment by Shineslippers** [comment]
Tested:

- [X] Created a dozen new characters and cancel the cinematic. Able to move with each new character after first login.

Anyway console still has error messages
```sql
Player Katrazha requests non-existing seer GUID Full: 0x0000000000000000 Type: None Low: 0
Player Buyag requests non-existing seer GUID Full: 0x0000000000000000 Type: None Low: 0
Player Vynleron requests non-existing seer GUID Full: 0x0000000000000000 Type: None Low: 0
```

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -12385,13 +12385,6 @@ void Player::SetViewpoint(WorldObject* target, bool apply)
 {
     if (apply)
     {
-        // target must be in world
-        if (!target->IsInWorld())
-            return;
-
-        if (!IsInWorld() || IsDuringRemoveFromWorld())
-            return;
```
Comment (acidmanifesto) [inline src/server/game/Entities/Player/Player.cpp:12393]: These will never happen.

**Comment by acidmanifesto** [comment]
> Tested:
> 
> * [x]  Created a dozen new characters and cancel the cinematic. Able to move with each new character after first login.
> 
> Anyway console still has error messages
> 
> ```sql
> Player Katrazha requests non-existing seer GUID Full: 0x0000000000000000 Type: None Low: 0
> Player Buyag requests non-existing seer GUID Full: 0x0000000000000000 Type: None Low: 0
> Player Vynleron requests non-existing seer GUID Full: 0x0000000000000000 Type: None Low: 0
> ```

Issue is fixed with this PR however i am marking it to be merged later until we can figure out exactly what that error message is for. I dont want to move to debug or remove any error logging without fully understanding the reason why it shows.

**Comment by acidmanifesto** [comment]
Per TC logging was moved to Debug, States its a Stating its a weird opcode handling issue which is not really known why.
Will be putting it back on to be merged status

**Pull request merged**